File: April/字典转模型/字典转模型/HJConnect.m
//
//  HJConnect.m
//  字典转模型
//
//  Created by liuhj on 16/4/5.
//  Copyright © 2016年 liuhj. All rights reserved.
//

#import "HJConnect.h"

@implementation HJConnect

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    
    if (self = [super init]) {
//        self.name = dict[@"name"];
//        self.telNum = dict[@"telNum"];
//        self.email = dict[@"email"];
//        self.addr = dict[@"addr"];
        [self setValuesForKeysWithDictionary:dict];
    }
    
    return self;
}

+ (instancetype)connectWithDictionary:(NSDictionary *)dict {
    
    return [[HJConnect alloc]initWithDictionary:dict];
    
}





@end

File: April/字典转模型-swift/字典转模型-swift/HJConnect.swift
//
//  HJConnect.swift
//  字典转模型-swift
//
//  Created by liuhj on 16/4/6.
//  Copyright © 2016年 liuhj. All rights reserved.
//

import UIKit

class HJConnect: NSObject {
    
    var name : String?
    var telNum : String?
    var email : String?
    var addr : String?
    
    func initWithDict(dict : NSDictionary) ->HJConnect {
        
        self.setValuesForKeysWithDictionary(dict as! [String : AnyObject])
        
        return self
        
    }
    
    class func connectWithDict(dict : NSDictionary) ->HJConnect {
        
        let conn = HJConnect()
        
        conn.initWithDict(dict)
        
        return conn
        
    }
    
    
    
    
    
}
